ui/app/src/App.tsx

```diff
@@ -93,14 +99,50 @@ function App() {
       >
         <CreateStream
           streamManagerAddress={config.streamManagerAddress as `0x${string}`}
-          amountPerSecond={100000000000000}
-          registerStream={(s: Stream) => console.log(s)}
+          amountPerSecond={100}
+          registerStream={(s: Stream) => {
+            console.log(s);
+            setStreamId(s.streamId);
+          }}
           renderReasonCode={renderReasonCode}
           handleTransactionStatus={handleTransactionStatus}
           tokenList={tokenList}
           cart={<Cart />}
         />
       </div>
+
+      <div className="status-graph">
+        <select
+          className="dropdown-select"
+          value={chartType}
+          onChange={(e) => setChartType(e.target.value as "bar" | "pie")}
+        >
+          <option value="bar">Bar Chart</option>
+          <option value="pie">Pie Chart</option>
+        </select>
+
+        <>
+          <StreamStatus
+            chartType={chartType}
+            stream={
+              new Stream(
+                config.streamManagerAddress as `0x${string}`,
+                address as `0x${string}`,
+                // todo: get streamID
+                streamId,
+                usePublicClient(),
+                useWalletClient()?.data as WalletClient
+              )
+            }
```

Comment:
Then this gets easier (just don't display the component while `stream` is null)
```suggestion
            stream={stream}
```